### ISSUE TITLE
Is there a way to not await completion of a batch?

### ISSUE THREAD

**Issue by lyio** [781flyingdutchman/background_downloader#12]
Basically, similar to how I can enqueue a single task and handle progress and status updates centrally via the listeners, can I do the same with a batch?

My use case requires me to track the single files individually anyway. But enqueuing them as a batch would make this a bit easier.

**Comment by 781flyingdutchman** [comment]
Yes, I think I can add that along with pause/resume functionality that I'm working on. Until then:
```
   for (var task in tasks) {
     FileDownloader().download(task, 
          onStatus: (status) {...},
          onProgress: (progress) {...}
     );
   }
```
might do the trick for a list of `tasks`. Just don't await the `download` call and keep track of batch completion yourself (or store the `Future` you get from the `download` call in a list and after the for loop wait for that list of futures to complete.)

**Comment by lyio** [comment]
Thanks for the answer. I am calling `enqueue` in a loop at the moment. I need the central tracking so I'll stick to that for now.

**Comment by 781flyingdutchman** [comment]
Implemented in version 5.0.0. You can now add a `batchProgressCallback` (the one that was there, but now it's a named parameter, which is why this is part of the breaking change to V5), and a task-specific `taskStatusCallback` and `taskProgressCallback`, which will be called for each task in the batch (just like a normal status or progress callback).

Pause/resume is now also available. Hope this helps!
